arraylist/src/main/java/ua/hillel/ArrayList.java

```diff
@@ -1,68 +1,125 @@
-
+        if (elements.length == size()) {
+            Object[] newElements = new Object[size() * 2];
+            System.arraycopy(element, 0, newElements, 0, size());
+            elements = newElements;
+        }
+        elements[size()] = element;
     }
 
     @Override
     public void add(int index, T element) {
-
+        if (elements.length == size()) {
+            Object[] newElements = new Object[size() * 2];
+            System.arraycopy(element, 0, newElements, 0, size());
+            elements = newElements;
+        }
+        elements[index] = element;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 ||index >= elements.length) {
+            throw new IndexOutOfBoundsException("Input correct index!");
+        }
+        return (T) elements[index];
     }
 
     @Override
     public T get(T element) {
-        return null;
+        if (element == null) {
+            throw new IllegalArgumentException("Input correct data!");
+        }
+
+        int index = -1;
+
+        for (int i = 0; i < elements.length; i++) {
+            if (elements[i].equals(element)) {
+                index = i;
+            }
+        }
+
+        if (index == -1) {
+            return null;
+        }
+        return (T) elements[index];
     }
 
     @Override
     public T getFirst() {
-        return null;
+        if (isEmpty()) {
+            return null;
+        }
+        return get(0);
     }
 
     @Override
     public T getLast() {
-        return null;
+        if (isEmpty()) {
+            return null;
+        }
+        return get(size() - 1);
     }
 
     @Override
     public void set(int index, T element) {
-
+        if ( index < 0 ||index >= elements.length) {
+            throw new IndexOutOfBoundsException("Input correct index!");
+        }
+        elements[index] = element;
     }
 
     @Override
-    public boolean remove(int index) {
-        return false;
+    public T remove(int index) {
+        if ( index < 0 ||index >= elements.length) {
+            throw new IndexOutOfBoundsException("Input correct index!");
+        }
+        T removedEl = (T) elements[index];
+
+        for (int i = 0; i < elements.length; i++) {
+            elements[i] = elements[i + 1];
+        }
+        elements[size() - 1] = null;
+        return removedEl;
     }
 
     @Override
     public boolean contains(T element) {
+        for (int i = 0; i < elements.length; i++) {
+            if (elements[i].equals(element)) {
+                return true;
+            }
+        }
         return false;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return elements.length == 0;
     }
 
     @Override
     public int size() {
-        return 0;
+        return elements.length;
     }
 
     @Override
     public void clear() {
-
+        for (int i = 0; i < elements.length; i++) {
+            remove(i);
+        }
```

Comment:
You should not call `remove()` here for each element. It is very slow and inefficient. The only thing you should do here is loop over the elements and set each one to null and then set size = 0; That's it, nothing else is required.